public/docs/_examples/cb-ts-to-js/js-es6/app/hero-di-inject-additional.component.es6

```diff
@@ -0,0 +1,61 @@
+import {
+  Attribute,
+  Component,
+  Inject,
+  Optional,
+  NgModule
+} from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
+// #docregion
+class TitleComponent {
+  constructor(titlePrefix, title) {
+    this.titlePrefix = titlePrefix;
+    this.title  = title;
+    this.msg = '';
+  }
+
+  ok() {
+    this.msg = 'OK!';
+  }
+}
+
```

Comment:
Oh wait, I see, you're still using `class` - then I don't think they are hoisted. Sorry, was getting confused between the ES6 and ES5 versions.